### PR TITLE
Add band_part function for tensorflow frontend

### PR DESCRIPTION
**Pull request by Change-m**
#4351

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 1]
```diff
@@ -394,3 +394,33 @@ def test_tensorflow_eye(
         batch_shape=batch_shape,
         dtype=dtype,
     )
+
+    
+# band_part
+    @handle_cmd_line_args
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:400]: Please fix the indentation.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @Change-m, I have suggested a change, please feel free to re-request my review once you're done. Thanks 🙂

**Comment by Change-m** [comment]
Hi [VedPatwardhan](https://github.com/VedPatwardhan), I fix the indentation.

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 2]
```diff
@@ -43,3 +43,7 @@ def eye(num_rows, num_columns=None, batch_shape=None, dtype=ivy.float32, name=No
 
 
 eye.unsupported_dtypes = {"torch": ("float16", "bfloat16")}
+
+
+def band_part(input, num_lower, num_upper, name=Nome):
+    return ivy.band_part(input, num_lower, num_upper)
```
Comment (VedPatwardhan) [inline ivy/functional/frontends/tensorflow/linalg.py:49]: I don't think there's a `band_part` function in Ivy's functional API, so either you'll have to implement the `tensorflow` `band_part` frontend as a composition of existing Ivy functions or extend Ivy's functional API by adding `ivy.band_part` in a separate PR referring to the [docs](https://lets-unify.ai/ivy/contributing/4_open_tasks.html#ivy-api-extensions)

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 3]
```diff
@@ -394,3 +394,33 @@ def test_tensorflow_eye(
         batch_shape=batch_shape,
         dtype=dtype,
     )
+
+    
+# band_part
+@handle_cmd_line_args
+@given(
+    dtype_and_input=_get_dtype_and_matrix(),
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:402]: You need to generate data for all parameters in the function, i.e. `input`, `num_lower`, `num_upper`, `name` in the `given` block and then pass them to the `test_frontend_function`

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @Change-m, thanks for making the changes. Just requested a few more changes. Thanks🙂

**Comment by VedPatwardhan** [comment]
I'll close this PR as discussed. Thanks 🙂

**Pull request closed without merging**